flytekit/image_spec/image_spec.py

```diff
@@ -85,6 +88,9 @@ def image_name(self) -> str:
     def _image_name(self) -> str:
         """Construct full image name with tag."""
         tag = calculate_hash_from_image_spec(self)
+        if self.tag_format:
+            tag = self.tag_format.format(spec_hash=spec_hash)
```

Comment:
```suggestion
            tag = self.tag_format.format(spec_hash=tag)
```

this should be ^, right?